scipy/stats/_distn_infrastructure.py

```diff
@@ -2312,7 +2312,9 @@ def fit(self, data, *args, **kwds):
         scale = kwds.pop('scale', start[-1])
         args += (loc, scale)
         x0, func, restore, args = self._reduce_func(args, kwds)
-
+        _reduce_func = kwds.pop('_reduce_func', None)
+        if _reduce_func is not None:
+            x0, func, restore = _reduce_func(x0, func, data=data)
```

Comment:
This is the only change to `scipy/stats/_distn_infrastructure.py`

Reply:
@WarrenWeckesser is still considering this.

Reply:
@swallan Can you just override `_reduce_func`?